crypto-bigint/src/uint/add_mod.rs

```diff
@@ -1,123 +1,97 @@
 //! [`UInt`] addition modulus operations.
 
-use super::UInt;
-
-macro_rules! impl_add_mod {
-    ($size:expr, $base_name:ident, $test_name:ident) => {
-        impl UInt<$size> {
-            /// Computes `a + b mod p` in constant time.
-            ///
-            /// Assumes `a` and `b` are `< p`.
-            pub const fn add_mod(&self, rhs: &Self, p: &Self) -> Self {
-                base::$base_name(&self, &rhs, &p)
-            }
+use crate::{Limb, UInt};
+
+impl<const LIMBS: usize> UInt<LIMBS> {
```

Comment:
Given that we do not have specialization yet, this now prevents me from adding optimizations for specific limb sizes, which are critical for optimal performance

Reply:
What if we have a generic `const fn` version that always works, and traits which are specific to limb sizes?

That's how all of the existing macros in this crate work anyway... they write trait impls.

Reply:
fair, that should work as well